--- origsrc/apr-1.2.12/config.layout	2004-11-24 16:51:51.000000000 -0600
+++ src/apr-1.2.12/config.layout	2008-04-29 21:42:53.203125000 -0500
@@ -229,3 +229,20 @@
     infodir:       ${exec_prefix}/share/info
     libsuffix:    -${APR_MAJOR_VERSION}
 </Layout>
+
+# Cygwin layout
+<Layout Cygwin>
+    prefix:        /usr
+    exec_prefix:   ${prefix}
+    bindir:        ${exec_prefix}/bin
+    sbindir:       ${exec_prefix}/bin
+    libdir:        ${exec_prefix}/lib
+    libexecdir:    ${exec_prefix}/apr-${APR_MAJOR_VERSION}/modules
+    mandir:        ${prefix}/share/man
+    sysconfdir:    /etc
+    datadir:       ${prefix}/share
+    installbuilddir: ${datadir}/apr-${APR_MAJOR_VERSION}-build
+    includedir:    ${prefix}/include/apr-${APR_MAJOR_VERSION}
+    localstatedir: /var
+    libsuffix:     -${APR_MAJOR_VERSION}
+</Layout>
--- origsrc/apr-1.2.12/configure.in	2007-11-03 09:13:08.000000000 -0500
+++ src/apr-1.2.12/configure.in	2008-04-29 22:06:47.359375000 -0500
@@ -8,6 +8,7 @@
 
 AC_INIT(build/apr_common.m4)
 AC_CONFIG_HEADER(include/arch/unix/apr_private.h)
+AC_CONFIG_MACRO_DIR(build)
 AC_CONFIG_AUX_DIR(build)
 
 dnl 
@@ -197,8 +198,8 @@
 
 if test "x$use_libtool" = "xyes"; then
       lt_compile='$(LIBTOOL) $(LTFLAGS) --mode=compile $(COMPILE) -o $@ -c $< && touch $@'
-      LT_VERSION="-version-info `$get_version libtool $version_hdr APR`"
-      link="\$(LIBTOOL) \$(LTFLAGS) --mode=link \$(LT_LDFLAGS) \$(COMPILE) ${LT_VERSION} \$(ALL_LDFLAGS) -o \$@"
+      lt_version="-version-info `$get_version libtool $version_hdr APR`"
+      link="\$(LIBTOOL) \$(LTFLAGS) --mode=link \$(LT_LDFLAGS) \$(COMPILE) ${lt_version} -no-undefined \$(ALL_LDFLAGS) -o \$@"
       so_ext='lo'
       lib_target='-rpath $(libdir) $(OBJECTS)'
       export_lib_target='-rpath \$(libdir) \$(OBJECTS)'
